lib/src/configuration.cc

```diff
@@ -127,105 +141,134 @@ HW::ParseResult Configuration::initialize(int argc, char *argv[],
     }
 
     if (parse_result == HW::ParseResult::OK) {
-        // No further processing or user interaction are required if
-        // the parsing outcome is HW::ParseResult::HELP or VERSION
-        config_file_ = HW::GetFlag<std::string>("config-file");
-
+        config_file_ = lth_file::tilde_expand(HW::GetFlag<std::string>("config-file"));
         if (!config_file_.empty()) {
             parseConfigFile();
         }
-
-        if (enable_logging) {
-            setupLogging();
-        }
-
-        validateAndNormalizeConfiguration();
-        setAgentConfiguration();
-
-        initialized_ = true;
+    } else {
+        // No further processing or user interaction are required if
+        // the parsing outcome is HW::ParseResult::HELP or VERSION
     }
 
     return parse_result;
 }
 
-void Configuration::setStartFunction(
-        std::function<int(std::vector<std::string>)> start_function) {
-    start_function_ = start_function;
-}
+static void validateLogDirPath(const std::string& logfile) {
+    auto logdir_path = fs::path(logfile).parent_path();
 
-void Configuration::validateAndNormalizeConfiguration() {
-    // determine which of your values must be initalised
-    if (HW::GetFlag<std::string>("broker-ws-uri").empty()) {
-        throw Configuration::UnconfiguredError{ "broker-ws-uri value must be defined" };
-    } else if (HW::GetFlag<std::string>("broker-ws-uri").find("wss://") != 0) {
-        throw Configuration::UnconfiguredError { "broker-ws-uri value must start with wss://" };
+    if (fs::exists(logdir_path)) {
+        if (!fs::is_directory(logdir_path)) {
+            throw Configuration::Error { "log directory is not a directory" };
```

Comment:
Can we make this error look more like the next one? "cannot write to x; parent directory is not a directory" or something like that.